oak_runtime/src/node/logger.rs

```diff
@@ -47,51 +47,66 @@ impl super::Node for LogNode {
         handle: oak_abi::Handle,
         _notify_receiver: oneshot::Receiver<()>,
     ) {
-        loop {
-            // An error indicates the Runtime is terminating. We ignore it here and keep trying to
-            // read in case a Wasm Node wants to emit remaining messages. We will return
-            // once the channel is closed.
-            //
-            // TODO(#1100): Investigate option for improving code clarity and not busy-waiting
-            // during runtime termination.
-            let _ = runtime.wait_on_channels(&[handle]);
+        'wait: loop {
+            let result = runtime.wait_on_channels(&[handle]);
 
-            match runtime.channel_read(handle) {
-                Ok(Some(message)) => match LogMessage::decode(&*message.bytes) {
-                    Ok(msg) => log!(
-                        target: &format!("{}:{}", msg.file, msg.line),
-                        to_level(msg.level),
-                        "{}",
-                        msg.message,
-                    ),
-                    Err(error) => {
-                        error!("{} Could not parse LogMessage: {}", self.node_name, error)
+            // Read all available log messages (even if the Runtime is terminating).
+            'read: loop {
+                match runtime.channel_read(handle) {
+                    Ok(Some(message)) => match LogMessage::decode(&*message.bytes) {
+                        Ok(msg) => {
+                            // Log messages that arrive from Oak applications over a logging channel
+                            // are controlled by IFC, and so need to be emitted independently of
+                            // whether the Runtime has been built with the `oak_debug` feature
+                            // enabled (and thus whether log! is connected up to anything or not).
+                            // So send to stderr.
+                            let now: chrono::DateTime<chrono::Utc> = chrono::Utc::now();
+                            let timestamp = now.format("%Y-%m-%dT%H:%M:%SZ");
```

Comment:
Maybe just use https://docs.rs/chrono/0.4.15/chrono/struct.DateTime.html#method.to_rfc3339_opts ?

Reply:
Thanks, I hadn't spotted that (the [top-level docs](https://docs.rs/chrono/0.4.15/chrono/index.html#formatting-and-parsing) only mention the `to_rfc3339()` method)